apps/mobile/src/libs/components/categories-form/categories-form.tsx

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+
+import { Button, Checkbox, View } from "~/libs/components/components";
+import {
+	useAppForm,
+	useCallback,
+	useEffect,
+	useFormController,
+} from "~/libs/hooks/hooks";
+import {
+	DEFAULT_ERROR_MESSAGE,
+	toastMessage,
+} from "~/libs/packages/toast-message/toast-message";
+import { globalStyles } from "~/libs/styles/styles";
+import {
+	type CategoryDto,
+	quizCategoriesValidationSchema,
+} from "~/packages/quiz/quiz";
+
+type Properties = {
+	categories: CategoryDto[];
+};
+
+const NO_CATEGORIES_SELECTED = 0;
+
+const CategoriesForm: React.FC<Properties> = ({ categories }) => {
+	const { control, errors, handleSubmit, setValue } = useAppForm({
+		defaultValues: {
+			categories: [] as number[],
+		},
+		validationSchema: quizCategoriesValidationSchema,
+	});
+
+	const { field } = useFormController({
+		control,
+		name: "categories",
+	});
+
+	const selectedCategories = field.value;
+
+	const onSubmit = (): void => {
+		const selectedLabels = categories
+			.filter((category) => selectedCategories.includes(category.id))
+			.map((category) => category.name)
+			.join(", ");
+
+		if (selectedLabels.length > NO_CATEGORIES_SELECTED) {
+			toastMessage.info({ message: `Selected Categories: ${selectedLabels}` });
+		}
+	};
+
+	const handlePress = (): void => {
+		void handleSubmit(onSubmit)();
+	};
+
+	const handleCheckboxChange = useCallback(
+		(categoryId: number): void => {
+			const updatedCategories = selectedCategories.includes(categoryId)
+				? selectedCategories.filter((id) => id !== categoryId)
+				: [...selectedCategories, categoryId];
+			setValue("categories", updatedCategories);
+		},
+		[selectedCategories, setValue],
+	);
+
+	const handleAllChange = useCallback(
+		(value: boolean): void => {
+			const updatedCategories = value
+				? categories.map((category) => category.id)
+				: [];
+			setValue("categories", updatedCategories);
+		},
+		[setValue, categories],
+	);
+
+	useEffect(() => {
+		if (errors.categories) {
+			toastMessage.error({
+				message: errors.categories.message || DEFAULT_ERROR_MESSAGE,
+			});
```

Comment:
We are not using toast messages to show form validation errors. Show it the same way as we do for regular input